clippy_lints/src/drop_forget_ref.rs

```diff
@@ -103,6 +102,75 @@ declare_clippy_lint! {
     "calls to `std::mem::forget` with a value that implements Copy"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for calls to `std::mem::drop` with a value that does not implement `Drop`.
+    ///
+    /// ### Why is this bad?
+    /// Calling `std::mem::drop` is no different than dropping such a type. A different value may
```

Comment:
Maybe the wording of this could be improved. It looks like it was copied and pasted from the `forget` lint and I think this description is confusing – yes `std::mem::drop` is in fact no different than dropping it, what’s the problem?